File: frontend/src/types/userAddress.ts
export interface UserAddress {
  id: string;
  fullName: string;
  phoneNumber: string;
  addressLine: string;
  ward?: string;
  district: string;
  province: string;
  postalCode?: string;
  addressType: string;
  isDefault: boolean;
  isActive: boolean;
  createdAt: string;
}

export interface CreateUserAddressRequest {
  fullName: string;
  phoneNumber: string;
  addressLine: string;
  ward?: string;
  district: string;
  province: string;
  postalCode?: string;
  addressType: string;
  isDefault: boolean;
}

export interface UpdateUserAddressRequest {
  fullName: string;
  phoneNumber: string;
  addressLine: string;
  ward?: string;
  district: string;
  province: string;
  postalCode?: string;
  addressType: string;
  isDefault: boolean;
}

export interface UserAddressResponse {
  success: boolean;
  message: string;
  address?: UserAddress;
  addresses?: UserAddress[];
  errors?: string[];
}

export interface AddressType {
  value: string;
  label: string;
  icon: string;
}

export const ADDRESS_TYPES: AddressType[] = [
  { value: 'Home', label: 'Nhà riêng', icon: '🏠' },
  { value: 'Office', label: 'Văn phòng', icon: '🏢' },
  { value: 'Other', label: 'Khác', icon: '📍' }
];

export interface Province {
  id: number;
  code: string;
  name: string;
  districts: District[];
}

export interface District {
  id: number;
  code: string;
  name: string;
  provinceId?: number;
  wards: Ward[];
}

export interface Ward {
  id: number;
  code: string;
  name: string;
  districtId?: number;
}

// Mock data removed - now using Viettel Post APIs for real-time address data

File: backend/Models/UserAddress.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Models
{
    public class UserAddress
    {
        [Key]
        public Guid Id { get; set; } = Guid.NewGuid();

        [Required]
        public string UserId { get; set; } = string.Empty;

        [Required]
        [StringLength(100)]
        public string FullName { get; set; } = string.Empty;

        [Required]
        [StringLength(15)]
        public string PhoneNumber { get; set; } = string.Empty;

        [Required]
        [StringLength(200)]
        public string AddressLine { get; set; } = string.Empty;


        [StringLength(100)]
        public string? Ward { get; set; } // Phường/Xã

        [Required]
        [StringLength(100)]
        public string District { get; set; } = string.Empty; // Quận/Huyện

        [Required]
        [StringLength(100)]
        public string Province { get; set; } = string.Empty; // Tỉnh/Thành phố

        // ✅ NEW: Store address IDs for shipping providers (ViettelPost, etc.)
        public int? ProvinceId { get; set; } // ViettelPost Province ID
        public int? DistrictId { get; set; } // ViettelPost District ID
        public int? WardId { get; set; } // ViettelPost Ward ID

        [StringLength(10)]
        public string? PostalCode { get; set; } // Mã bưu điện

        [StringLength(20)]
        public string AddressType { get; set; } = "Home"; // Home, Office, Other

        public bool IsDefault { get; set; } = false;

        public bool IsActive { get; set; } = true;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        // Navigation properties
        [ForeignKey("UserId")]
        public virtual User User { get; set; } = null!;
    }
}
